src/createClient.spec.ts

```diff
@@ -130,28 +130,93 @@ describe('query', () => {
 })
 
 describe('useQuery', () => {
-  testInEffectScope('responds to parameter changes', async () => {
-    const responseTrue = Symbol('responseTrue')
-    const responseFalse = Symbol('responseFalse')
+  describe('when parameters change', () => {
+    testInEffectScope('response is updated', async () => {
+      const responseTrue = Symbol('responseTrue')
+      const responseFalse = Symbol('responseFalse')
 
-    const action = vi.fn((value: boolean) => value ? responseTrue : responseFalse)
-    const { useQuery } = createClient()
+      const action = vi.fn((value: boolean) => value ? responseTrue : responseFalse)
+      const { useQuery } = createClient()
 
-    const input = ref(false)
+      const input = ref(false)
 
-    const query = useQuery(action, () => [input.value])
+      const query = useQuery(action, () => [input.value])
 
-    expect(query.response).toBe(undefined)
+      expect(query.response).toBe(undefined)
 
-    await nextTick()
+      await nextTick()
 
-    expect(query.response).toBe(responseFalse)
+      expect(query.response).toBe(responseFalse)
 
-    input.value = true
+      input.value = true
 
-    await nextTick()
+      await nextTick()
 
-    expect(query.response).toBe(responseTrue)
+      expect(query.response).toBe(responseTrue)
+    })
+
+    testInEffectScope('executed and executing are updated', async () => {
+      const responseTrue = Symbol('responseTrue')
+      const responseFalse = Symbol('responseFalse')
+
+      const action = vi.fn((value: boolean) => new Promise((resolve) => setTimeout(() => resolve(value ? responseTrue : responseFalse), 5000)))
+      const { useQuery } = createClient()
+
+      const input = ref(false)
+
+      const query = useQuery(action, () => [input.value])
+
+      await nextTick()
+
+      expect(query.executing).toBe(true)
+      expect(query.executed).toBe(false)
+
+      await vi.runAllTimersAsync()
+
+      expect(query.executing).toBe(false)
+      expect(query.executed).toBe(true)
+
+      input.value = true
+
+      await nextTick()
+
+      expect(query.executing).toBe(true)
+      expect(query.executed).toBe(false)
+
+      await vi.runAllTimersAsync()
+
+      expect(query.executing).toBe(false)
+      expect(query.executed).toBe(true)
+    })
+
+    testInEffectScope('when parameters become null, response is set to placeholder', async () => {
```

Comment:
I wrote this test to prove what I think was a bug that I fixed in createClient. When args become `null` we only Object.assigned the response. So executed and executing were still pointing to the previous query

Reply:
This will likely change. But I like having this test 👍